java/org/contikios/cooja/contikimote/ContikiMoteType.java

```diff
@@ -360,34 +353,15 @@ public File getExpectedFirmwareFile(String name) {
     return new File(new File(name).getParentFile(), "build/cooja/" + identifier + "." + getMoteType());
   }
 
-  /**
-   * Abstract base class for concrete section parser class.
-   */
-  interface SectionParser {
-    Map<String, Symbol> parseSymbols(Map<String, Symbol> inVars);
-  }
-
   /**
    * Parses Map file for section data.
    */
-  private static class MapSectionParser implements SectionParser {
-    private final String readelfData;
-
-    MapSectionParser(String readelfData) {
-      this.readelfData = readelfData;
-    }
-
-    @Override
-    public Map<String, Symbol> parseSymbols(Map<String, Symbol> inVars) {
+  private static class MapSectionParser {
+    static Map<String, Symbol> parseSymbols(String readelfData) {
```

Comment:
Is there any reason to keep these inner classes when they only contain a static method and no data?

Reply:
No, but the code should move elsewhere for efficiency reasons, so I made the smallest change with the current structure to simplify code review.